web/source/kmwembedded.ts

```diff
@@ -386,14 +386,19 @@ namespace com.keyman.text {
       osk.vkbd.keyPending = null;
 
       // Changes for Build 353 to resolve KMEI popup key issues      
-      keyName=keyName.replace('popup-',''); //remove popup prefix if present (unlikely)      
-      
-      var t=keyName.split('-'),layer=(t.length>1?t[0]:core.keyboardProcessor.layerId);
-      keyName=t[t.length-1];
+      keyName=keyName.replace('popup-',''); //remove popup prefix if present (unlikely)
+      // Can't just split on '-' because some layers like ctrl-shift contain it.
+      let separatorIndex = keyName.lastIndexOf('-');
+      var layer = core.keyboardProcessor.layerId;
+      if (separatorIndex > 0) {
+        layer = keyName.substring(0, separatorIndex);
```

Comment:
I originally had
`layer = keyName.substring(0, separatorIndex-1)`

but that was giving me layer `ctrl-al` 😕

Reply:
What you now have is right.  The index of a character in a string matches the length of the substring before it, after all.